Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfConversions.cs

```diff
@@ -0,0 +1,444 @@
+
+        // glTF matrix: column vectors, column-major storage, +Y up, +Z forward, -X right, right-handed
+        // unity matrix: column vectors, column-major storage, +Y up, +Z forward, +X right, left-handed
+        // multiply by a negative X scale to convert handedness
+        public static readonly Vector3 CoordinateSpaceConversionScale = new Vector3(-1, 1, 1);
+
+        public static readonly Vector4 TangentSpaceConversionScale = new Vector4(-1, 1, 1, -1);
+
+        public static Matrix4x4 GetTrsProperties(this GltfNode node, out Vector3 position, out Quaternion rotation, out Vector3 scale)
+        {
+            Matrix4x4 matrix = node.matrix.GetMatrix4X4Value();
+
+            if (!node.useTRS)
+            {
+                matrix.GetTrsProperties(out position, out rotation, out scale);
+            }
+            else
+            {
+                position = node.translation.GetVector3Value();
+                rotation = node.rotation.GetQuaternionValue();
+                scale = node.scale.GetVector3Value(false);
+            }
+
+            return matrix;
+        }
+
+        public static Color GetColorValue(this float[] colorArray)
+        {
+            return new Color(colorArray[0], colorArray[1], colorArray[2], colorArray.Length < 4 ? 1f : colorArray[3]);
+        }
+
+        public static float[] SetColorValue(this Color color)
+        {
+            return new[] { color.r, color.g, color.b, color.a };
+        }
+
+        public static Vector2 GetVector2Value(this float[] vector2Array)
+        {
+            return new Vector2(vector2Array[0], vector2Array[1]);
+        }
+
+        public static float[] SetVector2Value(this Vector2 vector)
+        {
+            return new[] { vector.x, vector.y };
+        }
+
+        public static Vector3 GetVector3Value(this float[] vector3Array, bool convert = true)
+        {
+            var vector = new Vector3(vector3Array[0], vector3Array[1], vector3Array[2]);
+            return convert ? Vector3.Scale(vector, CoordinateSpaceConversionScale) : vector;
+        }
+
+        public static float[] SetVector3Value(this Vector3 vector, bool convert = true)
+        {
+            if (convert)
+            {
+                vector = Vector3.Scale(vector, CoordinateSpaceConversionScale);
+            }
+
+            return new[] { vector.x, vector.y, vector.z };
+        }
+
+        public static Quaternion GetQuaternionValue(this float[] quaternionArray, bool convert = true)
+        {
+            var axes = new Vector3(quaternionArray[0], quaternionArray[1], quaternionArray[2]);
+
+            if (convert)
+            {
+                axes = Vector3.Scale(axes, CoordinateSpaceConversionScale) * -1.0f;
+            }
+
+            return new Quaternion(axes.x, axes.y, axes.z, quaternionArray[3]);
+        }
+
+        public static float[] SetQuaternionValue(this Quaternion quaternion, bool convert = true)
```

Comment:
nit: it may be worth adding summaries to these get/sets calling out that a get converts from gltf to unity coordinate space wheras a set converts from unity to gltf